src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/customize_panel/customize_panel_editor.tsx

```diff
@@ -125,8 +125,14 @@ export const CustomizePanelEditor = (props: CustomizePanelProps) => {
             <EuiButtonEmpty
               size="xs"
               data-test-subj="resetCustomEmbeddablePanelTitleButton"
-              onClick={() => setPanelTitle(embeddable.getOutput().defaultTitle)}
-              disabled={hideTitle || !editMode}
+              onClick={() => {
+                setPanelTitle(embeddable.getInput().title ?? embeddable.getOutput().defaultTitle);
+              }}
```

Comment:
This change should be undone: the reset button should always set it back to the default title.

```suggestion
              onClick={() => setPanelTitle(embeddable.getOutput().defaultTitle)}
```